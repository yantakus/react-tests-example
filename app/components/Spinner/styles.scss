@keyframes spinner {
  0%{ background-color: rgba(255, 255, 255, 1) }
  100%{ background-color: rgba(255, 255, 255, .2) }
}

.spinner {
  position: relative;
  height: 15px;
  width: 25px;
  margin: 1px auto 0;
  > div {
    background-color: rgba(255, 255, 255, .2);
    height: 6px;
    position: absolute;
    width: 3px;
    border-radius: 6px;
    animation-name: spinner;
    animation-duration: 1.04s;
    animation-iteration-count: infinite;
    &:nth-child(1) {
      left: 2px;
      top: 5px;
      transform: rotate(-90deg);
    }
    &:nth-child(2) {
      left: 4px;
      top: -2px;
      transform: rotate(-45deg);
      animation-delay: .13s;
    }
    &:nth-child(3) {
      left: 11px;
      top: -5px;
      transform: rotate(0deg);
      animation-delay: .26s;
    }
    &:nth-child(4) {
      right: 4px;
      top: -2px;
      transform: rotate(45deg);
      animation-delay: .39s;
    }
    &:nth-child(5) {
      right: 2px;
      top: 5px;
      transform: rotate(90deg);
      animation-delay: .52s;
    }
    &:nth-child(6) {
      right: 4px;
      bottom: -3px;
      transform: rotate(135deg);
      animation-delay: .65s;
    }
    &:nth-child(7) {
      bottom: -5px;
      left: 11px;
      transform: rotate(180deg);
      animation-delay: 1.78s;
    }
    &:nth-child(8) {
      left: 4px;
      bottom: -3px;
      transform: rotate(-135deg);
      animation-delay: .91s;
    }
  }
}
