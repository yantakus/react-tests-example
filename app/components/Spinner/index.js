// @flow

import React from 'react'
import './styles.scss'

const Spinner = () =>
  <div className='spinner'>
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>

export default Spinner
