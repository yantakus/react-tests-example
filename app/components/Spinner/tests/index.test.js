import React from 'react'
import renderer from 'react-test-renderer'
import Spinner from '../index'

test('Spinner matches the snapshot', () => {
  const component = renderer.create(
    <Spinner />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
