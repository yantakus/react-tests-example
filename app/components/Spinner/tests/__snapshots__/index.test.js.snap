exports[`test Spinner matches the snapshot 1`] = `
<div
  className="spinner">
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
